src/processing/aggregate_counts.py

```diff
@@ -0,0 +1,29 @@
+
+from pyspark.sql import functions as F
+
+def get_aggregate_counts(df, counting_col, alias_name):
```

Comment:
I am not a fan of just calling the dataframe df, it should be more descriptive (or at least input_df).

Also type hints and docstrings

Reply:
Normally I would agree but with this function, there is only one dataframe so I didn't want to cause any confusion if we needed to add to this. But we can change it if it makes more sense to?

Reply:
@humairahussein1 Linters might complain about a short variable name like this - even df_data, would be better, though it seems a bit redundant!